pkg/sql/types/types.proto

```diff
@@ -417,6 +417,21 @@ message GeoMetadata {
   optional geopb.Shape shape = 2 [(gogoproto.nullable)=false];
 }
 
+// SerializedUserDefinedTypeMetadata contains user defined type metadata
+// that will be serialized to disk, unlike other user defined type metadata
+// that is only stored in memory once a type is resolved.
+message SerializedUserDefinedTypeMetadata {
```

Comment:
minor nit: doesn't need `Serialized` as a prefix, as protobufs imply serialization.

Reply:
We already have a `UserDefinedTypeMetadata` for the in memory metadata, so I need some way to differentiate -- do you have a different suggestion?

Reply:
Ok i renamed to `PersistentUserDefinedTypeMetadata` -- thoughts?

Reply:
ah fair enough. works for me.